network/tailscale/tailscale.yaml

```diff
@@ -0,0 +1,73 @@
+name: tailscale
+depends:
+  - service: cri
+  - network:
+    - addresses
+    - connectivity
+    - etcfiles
+container:
+  entrypoint: ./containerboot
+  args: []
+  environment:
+  - PATH=/usr/local/bin:/sbin
+  - TS_SOCKET=/var/run/tailscale/tailscaled.sock
+  - TS_STATE_DIR=/var/lib/tailscale
+  - TS_USERSPACE=false
+  - TS_ACCEPT_DNS=true
```

Comment:
I think this maybe should be user configurable.
I might refactor the extension to read some configuration from a static file, including the TS_AUTH key so this would be more consistent over talos upgrades.

I want to do this before merging, as it's a significant change, so don't want someone to have a breaking change from this

Reply:
> I want to do this before merging, as it's a significant change, so don't want someone to have a breaking change from this

Cool, let us know when it's ready to be reviewed again, you could also mark the PR as a draft if it's still WIP

Reply:
Have changed this to use s6-envdir, so users can write MachineFiles into /var/etc/tailscale/env and set parts how they please